src/utils/include/fake_filesystem.hpp

```diff
@@ -0,0 +1,18 @@
+// A fake filesystem for cache httpfs extension testing purpose.
+
+#pragma once
+
+#include "duckdb/common/file_system.hpp"
+#include "duckdb/common/local_file_system.hpp"
+
+namespace duckdb {
+
+// WARNING: fake filesystem is used for testing purpose and shouldn't be used in production.
+class CacheHttpfsFakeFileSystem : public LocalFileSystem {
```

Comment:
Maybe FakeCacheHttpfs is more intuitive at first glance?

Reply:
All configurations and functions are prefixed with `cache_httpfs`.